DNN Platform/Library/Services/Upgrade/Upgrade.cs

```diff
@@ -3041,6 +3041,17 @@ private static void UpgradeToVersion740()
             ReIndexUserSearch();
         }
 
+		private static void UpgradeToVersion742()
+		{
+			var containerFolder = string.Format("{0}Containers\\DarkKnightMobile", Globals.HostMapPath);
+			var skinFolder = string.Format("{0}Skins\\DarkKnightMobile", Globals.HostMapPath);
+			if (!Directory.Exists(skinFolder) && !Directory.Exists(containerFolder))
+			{
+				UninstallPackage("DarkKnightMobile"); //Container
+				UninstallPackage("DarkKnightMobile"); //Skin
+			}

```

Comment:
These two checks should be done separately and the corresponding package removed.

If (!Directory.Exists(skinfolder)) { UninstallPackage("...") }
If (!Directory.Exists(containerFolder)) { UninstallPackage("...") }